core/src/main/java/com/graphhopper/matrix/algorithm/OneToOneLoopMatrixAlgorithm.java

```diff
@@ -0,0 +1,63 @@
+package com.graphhopper.matrix.algorithm;
+
+import com.graphhopper.matrix.DistanceMatrix;
+import com.graphhopper.routing.*;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.routing.util.TraversalMode;
+import com.graphhopper.routing.util.Weighting;
+import com.graphhopper.storage.Graph;
+
+import java.util.List;
+
+/**
+ * This implementation will just run a plain route algorithm for
+ * each origin-destination combination.
+ *
+ * Data structures between the route algorithms are not reused, resulting
+ * in O(R^2) performance hit, where R is the complexity of the underlying route algorithm,
+ * assuming a quadratic matrix (origins == destinations)
+ *
+ * For Dijkstra this is O( (|E|+|V|log(|V|))^2 )
+ *
+ * @author Pascal Büttiker
+ */
+public class OneToOneLoopMatrixAlgorithm extends AbstractMatrixAlgorithm {
+
+    private final RoutingAlgorithmFactory routingFactory = new RoutingAlgorithmFactorySimple();
+    private final AlgorithmOptions underlyingAlgo;
```

Comment:
naming 'underlyingAlgo' is misleading. Furthermore we can disable calculating instructions and paths in the hints